src/conn/conn_log.c

```diff
@@ -310,7 +310,13 @@ __wt_logmgr_config(WT_SESSION_IMPL *session, const char **cfg, bool reconfig)
      */
     WT_RET(__wt_config_gets(session, cfg, "log.prealloc", &cval));
     if (cval.val != 0)
-        conn->log_prealloc = 1;
+        if (cval.val != 0)
+            conn->log_prealloc = true;
+
+    WT_RET(__wt_config_gets(session, cfg, "log.min_preplog_file_count", &cval));
+    if (cval.val != 0)
+        conn->min_preplog_file_count = (uint32_t)cval.val;
+    conn->log_prealloc_count = conn->min_preplog_file_count;
```

Comment:
This can be simplified a lot by setting the initial value to your configuration setting only when enabled:
```
WT_RET(__wt_config_gets(session, cfg, "log.prealloc", &cval));
if (cval.val != 0) {
    WT_RET(__wt_config_gets(session, cfg, "log.prealloc_init_count", &cval));
    conn->log_prealloc = cval.val;
    WT_ASSERT(session, conn->log_prealloc > 0);
}
```
Then no other code changes are needed below.